src/app/file-manager/services/gcs-api.service.ts

```diff
@@ -24,13 +34,71 @@ export class GcsApiService extends GcsService {
   }
 
   public uploadFiles(bucketName: String, files: any[]) {
-      if (files !== null && files.length > 0) {
-        this.electronService.ipcRenderer.send('start-upload', bucketName, files, SecurityService.getAccessToken());
-      }
+    if (files !== null && files.length > 0) {
+      this.electronService.ipcRenderer.send('start-upload', bucketName, files, SecurityService.getAccessToken());
+    }
   }
 
   public downloadFiles(files: Item[]) {
     this.electronService.ipcRenderer.send('start-download', files, SecurityService.getAccessToken());
   }
 
+  public cancelAll() {
+    const itemsToDownload = this.getDownloadingFiles();
+    const itemsToUpload = this.getUploadingFiles();
+    if (itemsToDownload.length > 0) {
+      this.electronService.ipcRenderer.send('download-cancel');
+      this.cancelItemsStatus(itemsToDownload);
+    }
+    if (itemsToUpload.length > 0) {
+      this.electronService.ipcRenderer.send('upload-cancel');
+      this.cancelItemsStatus(itemsToUpload);
+    }
+  }
+
+  public cancelDownloads() {
+    const items = this.getDownloadingFiles();
+    if (items.length > 0) {
+      this.openModal('download-cancel', items, 'cancelAllDownloads');
+    }
+  }
+
+  public cancelUploads() {
+    const items = this.getUploadingFiles();
+    if (items.length > 0) {
+      this.openModal('upload-cancel', items, 'cancelAllUploads');
+    }
+  }
+
+  openModal(action: string, items: Item[], type: string) {
+    const dialogRef = this.dialog.open(WarningModalComponent, {
+      width: '500px',
+      disableClose: false,
+      data: type
+    });
+    dialogRef.afterClosed().subscribe(result => {
+      if (result.exit) {
+        this.electronService.ipcRenderer.send(action);
+        this.cancelItemsStatus(items);        }
+    });
+  }
+  cancelItemsStatus(items: Item []) {
+    items.forEach(item => {
+      this.store.dispatch(new Transferables.UpdateItemCanceled(item));
+    });
+  }
+
+  getDownloadingFiles() {
+    const items = new FilesDatabase(this.store).data.
+    filter(item => item.status === ItemStatus.DOWNLOADING ||
+      (item.status === ItemStatus.PENDING && item.type === Type.DOWNLOAD));
```

Comment:
Doesn't this duplicate the work in `TransferableState.downloadingCount()`? There should be a way to avoid the code duplication.

Reply:
This could cause a lot of changes, can we address this in the next sprint?

Reply:
OK